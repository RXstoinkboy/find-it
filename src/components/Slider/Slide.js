import React from 'react'
import styles from './styles/Slide.module.scss'
import PropTypes from 'prop-types'

export const Slide = props => {
    const inlineStyle = {
        backgroundImage: `url(${props.image.image})`,
        backgroundPosition: 'center',
        backgroundSize: 'cover',
        backgroundRepeat: 'no-repeat'
    }

    return (
        <>
            <div className={styles.slide} style={inlineStyle}>
                <a href={props.image.url} className={styles.link}>{props.image.name}</a>
            </div>
        </>
    );
};

export default Slide;

Slide.propTypes = {
    image: PropTypes.object.isRequired
}