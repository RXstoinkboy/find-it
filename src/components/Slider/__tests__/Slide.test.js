import React from 'react'
import {Slide} from '../Slide'
import toJson from 'enzyme-to-json'
import {shallow} from 'enzyme'

const imageMock = {
    url: 'some example url',
    name: 'hello example'
}

test(`Slide renders correctly`, ()=> {
    const slide = shallow(<Slide image={imageMock}/>);

    expect(toJson(slide)).toMatchSnapshot()
} )