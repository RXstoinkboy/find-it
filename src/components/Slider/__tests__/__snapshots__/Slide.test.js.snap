// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide renders correctly 1`] = `
<Fragment>
  <div
    className="slide"
    style={
      Object {
        "backgroundImage": "url(undefined)",
        "backgroundPosition": "center",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "cover",
      }
    }
  >
    <a
      className="link"
      href="some example url"
    >
      hello example
    </a>
  </div>
</Fragment>
`;
